### DISCUSSION TITLE
Defining Queries on the server

### DISCUSSION THREAD

**Discussion by joel-wenzel** [graphql-java/graphql-java#3429] [General]
I swore at one point I saw in the docs it was possible to define larger queries on the graphql server/schema. The client would just pass the query name, or something to that effect instead of the client sending the entire query string every time.

Does anyone know if that is possible

**Comment by dondonz** [comment]
Hello, you might be thinking of the concept of "Persisted Queries". It's not a GraphQL standard so you'll see different implementations around. In general, the idea is passing a hash which corresponds to a GraphQL operation or document which saves the amount of data to send.

You'll have to implement some sort of data store for this. You might be interested in these pages to learn more
* How Expedia did it: https://opensource.expediagroup.com/graphql-kotlin/docs/server/automatic-persisted-queries/
* Apollo: https://www.apollographql.com/docs/kotlin/advanced/persisted-queries/
* Relay support: https://relay.dev/docs/guides/persisted-queries/
